src/Table2Beta/Table.tsx

```diff
@@ -508,8 +508,19 @@ export class Table2Beta extends React.Component<Props, State> {
     }
 
     const { displayedData, numPages, allRows } = this._getDisplayedData();
-    selectedRows = new Set(selectedRows && allRows);
-    this.setState({ selectedRows });
+
+    // This occurs when a new external filter is applied.
+    // We want selectedRows to change if the filter affects the selection
+    const newSelectedRows = new Set(selectedRows);
+    newSelectedRows.forEach((item) => {
+      if (!allRows.includes(item)) {
+        newSelectedRows.delete(item);
+      }
+    });
+    if (newSelectedRows.size !== selectedRows.size) {
+      this.setState({ selectedRows: newSelectedRows });
+    }
```

Comment:
A quick note! A `setState` operation like this probably belongs in `componentDidMount` + `componentDidUpdate`. Calling `setState` within `render` is not recommended as `render` is not supposed to have any side effects. The only case where calling `setState` within `render` makes sense is, for example, as part of an `onClick` handler:

```tsx
render() {
  return (
    <button onClick={() => this.setState({ clicked: true })>Click here</button>
  );
}
```

Reply:
One reason why `setState` is advised against in render is because it can cause infinite loops as you've seen :smile:

Reply:
Good feedback! I did not know `render` was supposed to be side effect free 👍